source/MapPanel.cpp

```diff
@@ -808,10 +847,13 @@ void MapPanel::UpdateCache()
 					value = -1 + some + all;
 				}
 				else
+				{
 					value = SystemValue(&system);
+					UpdateColor(value);
+				}
 
 				if(colorSystem)
```

Comment:
I thought about it and I don't really see a better way of doing it, without changing the whole architecture, because colours are requested from another part; which is why the colour function and values are static; which as I understand is the problem here?
I made the color scale dynamic, dependant on what the max and min colours are and it is needed to know what those values are.

Reply:
Turns out I could just made them normal non static members, the other one using them inherits from this.
I hope its the same element seen as different classes though caus otherwise it won't work.